src/nedap/components/pedestal/service/component.clj

```diff
@@ -18,14 +19,19 @@
                                       :ssl? false}})
 
 (def dev-map
-  {::pedestal.http/join?           false
-   ::pedestal.http/allowed-origins {:creds true :allowed-origins (constantly true)}
+  {::pedestal.http/allowed-origins {:creds true :allowed-origins (constantly true)}
    ::pedestal.http/secure-headers  {:content-security-policy-settings {:object-src "'none'"}}})
 
 (speced/defn ^::service/component start [{{::router/keys [routes]} ::router/component
                                           ::service/keys           [defaults-kind pedestal-options expand-routes?]
                                           :as                      ^::service/initialized-component this}]
   (let [dev? (= :dev defaults-kind)
+
+        _ (when (and (not dev?)
+                     (not (spec/valid? ::service/pedestal-production-options pedestal-options)))
+            (throw (ex-info "Invalid :pedestal-options for production"
+                            (spec/explain-data ::service/pedestal-production-options pedestal-options))))
+
         ;; routes may be passed directly (via `::router/component`) or indirectly (via `pedestal-options`). Handle that:
```

Comment:
```suggestion
  (let [dev? (= :dev defaults-kind)]
        (when dev? (check! ::service/pedestal-production-options pedestal-options))
        ;; routes may be passed directly (via `::router/component`) or indirectly (via `pedestal-options`). Handle that:
```

Indentation might be off, also make sure to require `check!`

Reply:
good suggestion, somehow forgot about `check!`....